File: record-generator/processor/src/main/java/me/bristermitten/mittenlib/codegen/record/RecordSpec.java
package me.bristermitten.mittenlib.codegen.record;

import com.palantir.javapoet.ClassName;
import me.bristermitten.mittenlib.codegen.GenericTypeSpec;

public record RecordSpec(ClassName source, ClassName name, RecordConstructorSpec constructor)
        implements GenericTypeSpec, RecordSpecLike {}

File: record-generator/processor/src/main/scala/me/bristermitten/mittenlib/codegen/record/RecordSpec.scala
package me.bristermitten.mittenlib.codegen.record

import com.palantir.javapoet.ClassName
import me.bristermitten.mittenlib.codegen.GenericTypeSpec

case class RecordSpec(source: ClassName, name: ClassName, constructor: RecordConstructorSpec)
  extends GenericTypeSpec with RecordSpecLike
